RuntimeException: An error occured while executing doInBackground()

Posted by Oo Win Soe U on 26 Jul

```
USER_COMMENT=
ANDROID_VERSION=4.2.2
APP_VERSION_NAME=1.21
BRAND=Huawei
PHONE_MODEL=HUAWEI Y330-U11
CUSTOM_DATA=
STACK_TRACE=java.lang.RuntimeException: An error occured while executing doInBackground()
at android.os.AsyncTask$3.done(AsyncTask.java:299)
at java.util.concurrent.FutureTask.finishCompletion(FutureTask.java:352)
at java.util.concurrent.FutureTask.setException(FutureTask.java:219)
at java.util.concurrent.FutureTask.run(FutureTask.java:239)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
at java.lang.Thread.run(Thread.java:838)
Caused by: java.lang.NullPointerException
at fr.free.nrw.commons.upload.UploadController$2.doInBackground(UploadController.java:145)
at fr.free.nrw.commons.upload.UploadController$2.doInBackground(UploadController.java:112)
at android.os.AsyncTask$2.call(AsyncTask.java:287)
at java.util.concurrent.FutureTask.run(FutureTask.java:234)
... 3 more
java.lang.NullPointerException
at fr.free.nrw.commons.upload.UploadController$2.doInBackground(UploadController.java:145)
at fr.free.nrw.commons.upload.UploadController$2.doInBackground(UploadController.java:112)
at android.os.AsyncTask$2.call(AsyncTask.java:287)
at java.util.concurrent.FutureTask.run(FutureTask.java:234)
at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1080)
at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:573)
at java.lang.Thread.run(Thread.java:838)
```

--- a/app/src/main/java/fr/free/nrw/commons/upload/UploadController.java
+++ b/app/src/main/java/fr/free/nrw/commons/upload/UploadController.java
@@ -144,6 +144,7 @@ public class UploadController {
                 if(mimeType != null) {
                     contribution.setTag("mimeType", mimeType);
                     imagePrefix = mimeType.startsWith("image/");
+                    Log.d("UploadController", "MimeType is: " + mimeType);
                 }
 
                 if(imagePrefix && contribution.getDateCreated() == null) {

--- a/app/src/main/java/fr/free/nrw/commons/upload/UploadController.java
+++ b/app/src/main/java/fr/free/nrw/commons/upload/UploadController.java
@@ -139,10 +139,11 @@ public class UploadController {
 
                 if(mimeType == null || TextUtils.isEmpty(mimeType) || mimeType.endsWith("*")) {
                     mimeType = activity.getContentResolver().getType(contribution.getLocalUri());
-                    if(mimeType != null) {
-                        contribution.setTag("mimeType", mimeType);
-                        imagePrefix = mimeType.startsWith("image/");
-                    }
+                }
+
+                if(mimeType != null) {
+                    contribution.setTag("mimeType", mimeType);
+                    imagePrefix = mimeType.startsWith("image/");
                 }
 
                 if(imagePrefix && contribution.getDateCreated() == null) {

--- a/app/src/main/java/fr/free/nrw/commons/upload/UploadController.java
+++ b/app/src/main/java/fr/free/nrw/commons/upload/UploadController.java
@@ -135,14 +135,17 @@ public class UploadController {
                 }
 
                 String mimeType = (String)contribution.getTag("mimeType");
+                Boolean imagePrefix = false;
+
                 if(mimeType == null || TextUtils.isEmpty(mimeType) || mimeType.endsWith("*")) {
                     mimeType = activity.getContentResolver().getType(contribution.getLocalUri());
                     if(mimeType != null) {
                         contribution.setTag("mimeType", mimeType);
+                        imagePrefix = mimeType.startsWith("image/");
                     }
                 }
 
-                if(mimeType.startsWith("image/") && contribution.getDateCreated() == null) {
+                if(imagePrefix && contribution.getDateCreated() == null) {
                     Cursor cursor = activity.getContentResolver().query(contribution.getLocalUri(),
                             new String[]{MediaStore.Images.ImageColumns.DATE_TAKEN}, null, null, null);
                     if(cursor != null && cursor.getCount() != 0) {
@@ -159,7 +162,6 @@ public class UploadController {
                         contribution.setDateCreated(new Date());
                     }
                 }
-
                 return contribution;
             }
 

